tests/e2e/tidbcluster/serial.go

```diff
@@ -0,0 +1,112 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.package spec
+
+package tidbcluster
+
+import (
+	"context"
+	_ "net/http/pprof"
+
+	"github.com/onsi/ginkgo"
+	asclientset "github.com/pingcap/advanced-statefulset/pkg/client/clientset/versioned"
+	"github.com/pingcap/tidb-operator/pkg/client/clientset/versioned"
+	"github.com/pingcap/tidb-operator/tests"
+	e2econfig "github.com/pingcap/tidb-operator/tests/e2e/config"
+	"github.com/pingcap/tidb-operator/tests/e2e/util/portforward"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
+	f := framework.NewDefaultFramework("serial")
+
+	var ns string
+	var c clientset.Interface
+	var cli versioned.Interface
+	var asCli asclientset.Interface
+	var cfg *tests.Config
+	var config *restclient.Config
+	var fw portforward.PortForward
+	var fwCancel context.CancelFunc
+
+	ginkgo.BeforeEach(func() {
+		ns = f.Namespace.Name
+		c = f.ClientSet
+		var err error
+		config, err = framework.LoadConfig()
+		framework.ExpectNoError(err, "failed to load config")
+		cli, err = versioned.NewForConfig(config)
+		framework.ExpectNoError(err, "failed to create clientset")
+		asCli, err = asclientset.NewForConfig(config)
+		framework.ExpectNoError(err, "failed to create clientset")
+		clientRawConfig, err := e2econfig.LoadClientRawConfig()
+		framework.ExpectNoError(err, "failed to load raw config")
+		ctx, cancel := context.WithCancel(context.Background())
+		fw, err = portforward.NewPortForwarder(ctx, e2econfig.NewSimpleRESTClientGetter(clientRawConfig))
+		framework.ExpectNoError(err, "failed to create port forwarder")
+		fwCancel = cancel
+		cfg = e2econfig.TestConfig
+	})
+
+	ginkgo.AfterEach(func() {
+		if fwCancel != nil {
+			fwCancel()
+		}
+	})
+
+	// tidb-operator with AdvancedStatefulSet feature enabled
+	ginkgo.Context("[Feature: AdvancedStatefulSet]", func() {
+		var ocfg *tests.OperatorConfig
+		var oa tests.OperatorActions
+
+		ginkgo.BeforeEach(func() {
+			ocfg = &tests.OperatorConfig{
+				Namespace:      "pingcap",
+				ReleaseName:    "operator",
+				Image:          cfg.OperatorImage,
+				Tag:            cfg.OperatorTag,
+				SchedulerImage: "k8s.gcr.io/kube-scheduler",
+				Features: []string{
+					"StableScheduling=true",
+					"AdvancedStatefulSet=true",
+				},
+				LogLevel:        "4",
+				ImagePullPolicy: v1.PullIfNotPresent,
+				TestMode:        true,
+			}
+			oa = tests.NewOperatorActions(cli, c, asCli, tests.DefaultPollInterval, ocfg, e2econfig.TestConfig, nil, fw, f)
+			ginkgo.By("Installing CRDs")
+			oa.CleanCRDOrDie()
+			oa.InstallCRDOrDie()
+			ginkgo.By("Installing tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
+			oa.DeployOperatorOrDie(ocfg)
```

Comment:
Do we need to clean and install `tidb-operator` in every "spec"?

Reply:
yes, serial tests are tests that want to test with their own tidb-operator. 

Actually, this deploy test with AdvancedStatefulSet feature can run against a shared tidb-operator with AdvancedStatefulSet feature in parallel. Before we have a job to run specs against tidb-operator with AdvancedStatefulSet. We can test it in the serial job.